### PR TITLE
Add job to try and build gpu package in CI

### PR DESCRIPTION
**Pull request by mtreinish**
### Summary

To avoid packaging issues we should automate the building the
qiskit-aer-gpu packages like what is done for the non-gpu packages. A
GPU shouldn't be required because the compilation with nvcc doesn't
actually run on a GPU. It is just a matter of figuring out how to get
nvcc installed and configured in the manylinux docker image. This commit
starts that process.

### Details and comments

### CODE REVIEW

**Comment by mtreinish** [comment]
This should be good to go now. You can see the last build here: https://github.com/mtreinish/qiskit-aer/suites/856091601/artifacts/9809258 (I'll post a new link after the last run since that's about a month old). Since this is a fork PR gha is running it on my fork instead of on the PR. After the next run I'll change the job to only run on tags and remove the WIP.

**Comment by mtreinish** [comment]
You can see the output for the build here: https://github.com/mtreinish/qiskit-aer/runs/922962679?check_suite_focus=true I'll update with a link to artifacts zip file with the wheels in after the job finishes

**Comment by mtreinish** [comment]
The results from the recent build are here: https://github.com/mtreinish/qiskit-aer/suites/978978671/artifacts/12504534 this includes py35 and py36 wheels. (I forgot that py37 and py38 were skipped when I moved this from azure)

**Comment by atilag** [comment]
The generated artifacts worked for me locally.

**Comment by mtreinish** [comment]
Ok, this should be ready to go now, I've updated this to only trigger on tags and added a twine push step after the wheel builds.

**Comment by chriseclectic** [comment]
I tested the wheel on my ubuntu system with CUDA 10.1 and it was incompatible. Is it possible to make it work with 10.1 as a minimum version for the wheel?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.7'
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel==1.5.5
+      - name: Build wheels
+        env:
+          CIBW_BEFORE_ALL: "yum install -y yum-utils wget && wget https://developer.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda-repo-rhel6-10-1-local-10.1.105-418.39-1.0-1.x86_64.rpm && rpm -i cuda-repo-rhel6-10-1-local-10.1.105-418.39-1.0-1.x86_64.rpm && yum clean all && yum -y install cuda"
```
Comment (chriseclectic) [inline .github/workflows/main.yml:21]: Should  this be:
```suggestion
          CIBW_BEFORE_ALL: "yum install -y yum-utils wget && wget https://developer.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda-repo-rhel6-10-1-local-10.1.105-418.39-1.0-1.x86_64.rpm && rpm -i cuda-repo-rhel6-10-1-local-10.1.105-418.39-1.0-1.x86_64.rpm && yum clean all && yum -y install cuda-10.1"
```

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+      - uses: actions/setup-python@v2
+        name: Install Python
+        with:
+          python-version: '3.7'
+      - name: Install cibuildwheel
+        run: |
+          python -m pip install cibuildwheel==1.5.5
+      - name: Build wheels
+        env:
+          CIBW_BEFORE_ALL: "yum install -y yum-utils wget && wget http://developer.download.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda-repo-rhel6-10-1-local-10.1.243-418.87.00-1.0-1.x86_64.rpm && rpm -i cuda-repo-rhel6-10-1-local-10.1.243-418.87.00-1.0-1.x86_64.rpm && yum clean all && yum -y install cuda-10-1"
```
Comment (chriseclectic) [inline .github/workflows/main.yml:21]:
```suggestion
          CIBW_BEFORE_ALL: "yum install -y yum-utils wget && wget https://developer.download.nvidia.com/compute/cuda/10.1/Prod/local_installers/cuda-repo-rhel6-10-1-local-10.1.243-418.87.00-1.0-1.x86_64.rpm && rpm -i cuda-repo-rhel6-10-1-local-10.1.243-418.87.00-1.0-1.x86_64.rpm && yum clean all && yum -y install cuda-10-1"
```

**Comment by mtreinish** [comment]
I should have tagged this as `on hold` it wasn't ready to merge yet because https://github.com/Qiskit/qiskit-aer/pull/775/commits/efbf60be2ffabbfeb80451092a30dca42e440b30 was applied on the branch for testing. With this merged as is we'll be triggering a GPU wheel build for ever pull request and merged commit. It also doesn't push to pypi either. I'll open up a follow up pr about this

**Pull request merged**